test/suites/integration/deregister_targets_test.go

```diff
@@ -68,7 +63,11 @@ var _ = Describe("Deregister Targets", func() {
 	})
 
 	AfterEach(func() {
-		testFramework.CleanTestEnvironment(ctx)
+		// Lattice targets draining time for test cases in this file is actually unavoidable,
+		//because these test cases logic themselves test lattice targets de-registering before delete the httproute
+		testFramework.ExpectDeleted(ctx, gateway, pathMatchHttpRoute)
+		time.Sleep(30 * time.Second) // Use a trick to delete httpRoute first and then delete the service and deployment to avoid draining lattice targets
+		testFramework.ExpectDeleted(ctx, service, deployment)
```

Comment:
Can you explain (probably again) why do you need to use trick for 30 seconds wait? And whats the deal to not use trick?

Reply:
Some backgroud: If a latticeService's listener's rule can route to that targetGroup, targets in that tg will have `healthy`/`unhealthy` status, derigister a healthy status target can cause  5 min draining. if no rule can route to that targetGroup, the target would be `unused` status, we coud immediately de-register it without draining time.


Some other backgroud: when delete a httproute, it following this order to delete : latticeService -> listener -> rules -> targets -> target group that is pretty streamline, target status chagne to  `unused` first, and then could be immediately deleted.

However, if we try to delete a k8sDeplotment or k8sService prior than deleting a httproute, controller will try to de-register a `healthy` status target, that cause draining status targets

Reply:
interesting, should we change controller's logic to removing route prior to registration?